lib/hobo_fields/model.rb

```diff
@@ -50,7 +50,8 @@ def self.inherited(klass)
     module ClassMethods
       def index(fields, options = {})
         # don't double-index fields
-        unless index_specs.*.fields.include?(Array.wrap(fields).*.to_s)
+        index_fields_s = Array.wrap(fields).*.to_s
+        unless index_specs.any? { |index_spec| index_spec.fields == index_fields_s }
```

Comment:
`.*.include?` is an anti-pattern: it has to build the interim array completely first...with no short-circuiting. `.any?` short-circuits.